File: Domain/Entities/HairSalon.cs
using System.Collections.Generic;

namespace Domain.Entities
{
    public class HairSalon
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public string Description { get; set; }
        public bool IsPayed { get; set; }
        public bool IsPremium { get; set; }
        public int CityId { get; set; }

        public virtual ICollection<WorkingHour> WorkingHours { get; set; }
        public virtual ICollection<Service> Services { get; set; }
        public virtual ICollection<Review> Reviews { get; set; }
        public virtual ICollection<Reservation> Reservations { get; set; }
        public virtual ICollection<Favourite> Favourites { get; set; }
        public virtual ICollection<Image> Images { get; set; }

        public virtual City City { get; set; }
    }
}

File: WebUI/client-app/src/app/models/hairSalon.ts
export interface IHairSalonsVm {
  hairSalons: IHairSalon[];
  hairSalonsCount: number;
}

export interface IHairSalon {
  id?: number;
  name: string;
  email: string;
  address: string;
  phone: string;
  description: string;
  isPayed: boolean;
  isPremium: boolean;
  isFavourite: boolean;
  isReviewed: boolean;
  city: ICity;
  image: IImage;
  workingHours: IWorkingHour[];
  reviews: IReview[];
  images: IImage[];
  services: IService[];
}

interface ICity {
  id: number;
  name: string;
}

interface IWorkingHour {
  id: number;
  day: number;
  openTime: string;
  closeTime: string;
}

interface IReview {
  user: IUser;
  rating: number;
  comment: string;
  createdAt: Date;
}

interface IUser {
  id: string;
  username: string;
}

interface IImage {
  id: string;
  url: string;
  isMain: boolean;
}

interface IService {
  id: number;
  name: string;
  price: number;
}
